File: ruby/lib/user_interface/user_input.rb
module Ui

public class UserInput

	char command
	public String[] args

	UserInput(String line)
		command = line.charAt(0)
		args = line.split("\\s+")
	end

	public String args(int i)
		if (i < 0)
			throw new IllegalArgumentException
		return (i < args.length) ? args[i] : null
	end

end

File: java/src/com/xpsurgery/cornershop/ui/UserInput.java
package com.xpsurgery.cornershop.ui;

public class UserInput {

	char command;
	public String[] args;

	UserInput(String line) {
		command = line.charAt(0);
		args = line.split("\\s+");
	}

	public String args(int i) {
		if (i < 0)
			throw new IllegalArgumentException();
		return (i < args.length) ? args[i] : null;
	}

}
